### ISSUE TITLE
bug(Style): Enabling `preserveSymlinks` breaks stylesheet

### ISSUE THREAD

**Issue by myrddraall** [angular/angular-cli#25122]
### Is this a regression?

- [X] Yes, this behavior used to work in the previous version

### The previous version in which this bug was not present was

13

### Description

when enabling `preserveSymlinks` in angular.json custom stylesheets no longer compile because they cannot find `@material/button/button`

### Reproduction

Steps to reproduce:
1.  Run `ng new MyProject`
2. Select the default options
3. Run `cd MyProject`
4. `ng add @angular/material`
5. Select the default options, except for the theme, select Custom
6. Set  `preserveSymlinks` to true in angular.json
7.  Run `ng build`

### Expected Behavior

For the project to build without error

### Actual Behavior

The build fails with the following message:
```
./src/styles.scss - Error: Module build failed (from ./node_modules/.pnpm/sass-loader@13.2.0_sass@1.57.1+webpack@5.75.0/node_modules/sass-loader/dist/cjs.js):
SassError: Can't find stylesheet to import.
  ╷
2 │ @use '@material/button/button' as mdc-button;
  │ ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  ╵
  node_modules/@angular/material/button/_button-theme.scss 2:1               @use
  node_modules/@angular/material/core/density/private/_all-density.scss 2:1  @forward
  @angular/_index.scss 18:1                                                  @use
  src/styles.scss 1:1                                                        root stylesheet

```

### Environment

Angular CLI: 15.1.4
Node: 18.13.0
Package Manager: pnpm 7.26.2
OS: linux x64

Angular: 15.1.3
... animations, cdk, common, compiler, compiler-cli, core, forms
... material, material-date-fns-adapter, platform-browser
... platform-browser-dynamic, router

Package                         Version
---------------------------------------------------------
@angular-devkit/architect       0.1501.4 (cli-only)
@angular-devkit/build-angular   15.1.4
@angular-devkit/core            15.1.4 (cli-only)
@angular-devkit/schematics      15.1.4 (cli-only)
@angular/cli                    15.1.4
@schematics/angular             15.1.4 (cli-only)
rxjs                            7.5.6
typescript                      4.9.5

**Comment by alan-agius4** [comment]
Thanks for reporting this issue. This issue is now obsolete due to changes in the recent releases. Please update to the most recent Angular CLI version.

If the problem persists after upgrading, please open a new issue, provide a simple repository reproducing the problem, and describe the difference between the expected and current behavior.

**Comment by bruno-fonts** [comment]
Hi @alan-agius4, thanks for the update on this matter and for fixing this. Would you mind pointing us to the CLI version that should have resolved this? Would this be this latest `17.2.0` or already fixed somewhere in v16? Thanks 🙏
